src/ui.rs

```diff
@@ -165,8 +165,7 @@ impl Ui {
         let term = Term::stdout();
         let mut stage = Stage::new(term.clone(), &heading(&target, 1), None);
         let mut stages = 0;
-        let mut errors = 0;
-        let mut rc: i32 = 0;
+        let mut rc = Some(0);
 
         // Main state machine loop
```

Comment:
Would it be better to abstract the state machine in its own function which would return a `Result<i32>`, or make `rc` a `Result<i32>` in the first place, so the error can be annotated  where it happens and bubbled up out of `target_ui`.

Reply:
Refactoring in a function may not be the cleanest option with the code the way it is. The loop needs to modify local var `stage` which has some destructor semantics attached to it. Result<i32> has some error copy/clone issues. Originally I was trying to make Result<i32> work but `anyhow` was giving me a rude time. 

So for now I think what we have is the cleanest option with minimal changes.